maven-plugin/src/main/java/org/openmrs/maven/plugins/Pull.java

```
@@ -0,0 +1,293 @@
+            }
+        }else {
+            wizard.showMessage(String.format(NO_WATCHED_MODULES_MESSAGE, serverId));
+        }
+    }
+
+    /**
+     * Pull latest changes from upstream master
+     *
+     * @param path
+     * @return true if pull was successful
+     * @throws MojoExecutionException
+     */
+    private boolean pullLatestUpstream(String path) throws MojoExecutionException {
+        Repository localRepo;
+        Git git = null;
+        String previousBranch;
+        String newBranchFull;
+        String newBranch = "sdk-" + serverId;
+        try {
+            localRepo = new FileRepository(new File(path, ".git").getAbsolutePath());
+            git = new Git(localRepo);
+
+            previousBranch = localRepo.getBranch();
+
+            checkoutAndCreateNewBranch(git, newBranch);
+            newBranchFull = localRepo.getFullBranch();
+            RevCommit stash = git.stashCreate().call();
+            addRemoteTempRepo(git, path, newBranch);
+            PullResult pullResult = pullFromRemote(git, newBranch);
+            if(!pullResult.isSuccessful()){
+                rebaseAbort(git);
+                checkoutBranch(git, previousBranch);
+                failureReason = CONFLICT_MESSAGE;
+                return false;
+            }
+            if (stash != null) {
+                try {
+                    git.stashApply().setStashRef(stash.getName()).call();
+                    checkoutBranch(git, previousBranch);
+                    mergeWithNewBranch(git, newBranchFull);
+                    return true;
+                } catch (StashApplyFailureException e) {
+                    git.reset().setMode(ResetCommand.ResetType.HARD).setRef(previousBranch).call();
+                    checkoutBranch(git, previousBranch);
+                    git.stashApply().setStashRef(stash.getName()).call();
+                    failureReason = CONFLICT_MESSAGE;
```

It's better to throw a checked exception with your message and also passing the original exception.